File: swea/D2/1961.숫자 배열 회전/1961.숫자 배열 회전.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<vector<int>> turn90(vector<vector<int>> arr){
  int n = arr.size();

  vector<vector<int>> r_arr(n, vector<int>(n));

  for(int i = 0; i < n; i++){
    for(int j = 0; j < n; j++){
      r_arr[i][j] = arr[n-j-1][i];
    }
  }

  return r_arr;
}

int T;

int main(){
  cin >> T;

  int n;
  for(int t = 1; t <= T; t++){
    cin >> n;
    vector<vector<int>> arr(n, vector<int>(n));
    
    for(int i = 0; i < n; i++){
      for(int j = 0; j < n; j++){
        cin >> arr[i][j];
      }
    }

    auto arr90 = turn90(arr);
    auto arr180 = turn90(arr90);
    auto arr270 = turn90(arr180);

    cout << '#' << t << endl;

    for(int x = 0; x < n; x++){
      for(int y = 0; y < n; y++){
        cout << arr90[x][y];
      }
      cout << ' ';
      
      for(int y = 0; y < n; y++){
        cout << arr180[x][y];
      }
      cout << ' ';
      
      for(int y = 0; y < n; y++){
        cout << arr270[x][y];
      }
      cout << '\n';
    }
  }
}

File: swea/D2/1961.숫자 배열 회전/1961.숫자 배열 회전.py
def t_90(arr, n):
    t_arr = [[] for _ in range(n)]
    for i in range(n):
        for j in range(n-1,-1,-1):
            t_arr[i].append(arr[j][i])
    return t_arr

T = int(input())

for a in range(1,T+1):
    n = int(input())
    arr = [list(map(int,input().split())) for _ in range(n)]
    
    arr_90 = t_90(arr, n)
    arr_180 = t_90(arr_90, n)
    arr_270 = t_90(arr_180, n)

    print(f'#{a}')
    for i in range(n):
        print("".join(map(str, arr_90[i])),end=" ")
        print("".join(map(str, arr_180[i])),end=" ")
        print("".join(map(str, arr_270[i])),end="\n")